cmd/frontend/backend/external_services.go

```
@@ -89,36 +89,38 @@ func (e *externalServices) ExcludeRepoFromExternalService(ctx context.Context, e
 		return err
 	}
 
-	externalServices := tx.ExternalServices()
-	externalService, err := externalServices.GetByID(ctx, externalServiceID)
+	extSvcStore := tx.ExternalServices()
+	externalServices, err := extSvcStore.List(ctx, database.ExternalServicesListOptions{IDs: externalServiceIDs})
 	if err != nil {
 		return err
 	}
 
-	logger := e.logger.Scoped("ExcludeRepoFromExternalService", "excluding a repo from external service config").With(
-		log.Int64("externalServiceID", externalServiceID),
-		log.Int32("repoID", int32(repoID)),
-	)
-
-	// If external service doesn't support repo exclusion, then return.
-	if !externalService.SupportsRepoExclusion() {
-		logger.Warn("external service does not support repo exclusion")
-		return nil
+	logger := e.logger.Scoped("ExcludeRepoFromExternalServices", "excluding a repo from external service config").With(log.Int32("repoID", int32(repoID)))
+	for _, extSvcID := range externalServiceIDs {
+		logger = logger.With(log.Int64("externalServiceID", extSvcID))
 	}
 
-	repository, err := tx.Repos().Get(ctx, repoID)
-	if err != nil {
-		return err
+	for _, externalService := range externalServices {
+		// If external service doesn't support repo exclusion, then return.
+		if !externalService.SupportsRepoExclusion() {
+			logger.Warn("external service does not support repo exclusion")
+			return nil
+		}
 	}
```

Two independent questions here:

1. Should we return a custom error instead of logging a warning? 
2. Should we be skipping all the other external services if one of them does not supports repo exclusion? Won't it add to confusion as to what happened although the user just sees one warning log?

---

1) this is a very specific endpoint used only from one UI page which actually prevents this API from being called for code hosts without exclusion support.
I don't think we should return an error here as there are no consumers of this type of error at the moment (and 99% won't be in future as well).

2) as I've mentioned, this shouldn't really happen, but if it happens, I prefer to skip the updates all together than updating some code hosts and not updating other.